src/raft_server.c

```diff
@@ -1287,6 +1300,8 @@ void raft_pop_log(raft_server_t* me_, raft_entry_t* entries,
         raft_node_t* node = raft_get_node(me_, node_id);
         int is_self = node_id == raft_get_nodeid(me_);
 
+        assert(NULL != node);
```

Comment:
How about moving this assert call to line 1288/1301, before the is_self line?

Also, a similar comment as the one in raft_offer_log.